activejob/lib/active_job/enqueue_after_transaction_commit.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ActiveJob
+  module EnqueueAfterTransactionCommit # :nodoc:
+    extend ActiveSupport::Concern
+
+    included do
+      ##
+      # :singleton-method:
+      #
+      # Defines if enqueueing this job from inside an Active Record transaction
+      # automatically defers the enqueue to after the transaction commit.
+      #
+      # It can be set on a per job basis:
+      #  - `:always` forces the job to be deferred.
+      #  - `:never` forces the job to be queueed immediately
+      #  - `:auto` uses the queue adapter default behavior (recommended).
```

Comment:
I wonder if `:auto` should be `:adapter` or even `:default`. I think `:auto` to me implies that there is some decisive logic, whereas this is just a delegation.